app/mattermost.js

```diff
@@ -60,6 +62,10 @@ const launchApp = (credentials) => {
     EphemeralStore.appStarted = true;
 
     Linking.getInitialURL().then((url) => {
+        const serverUrl = urlParse(url).host;
+        if (serverUrl) {
+            store.dispatch(setServerUrl('https://' + serverUrl));
```

Comment:
although this seems to be what we want there are a few of things to consider.

* how can you be sure that the protocol has or requires SSL?
* the server url should only be set if the user is not logged in, imagine a user that is logged in to a different server, opens the deep link and now the server url changed.
* What if the app is configured to not allow server url changes?

Perhaps the best approach here is to get the initial URL in the serverUrl screen to avoid all of this and only set it when it applies

Reply:
hey @enahum, thanks for the heads up. I am fixing that.

Reply:
hey @enahum, thanks for the heads up. I am fixing that.